File: By-Swift/By-Swift/Leetcode/valid-parentheses.swift
class Solution {
    func isValid(_ s: String) -> Bool {
        guard !s.isEmpty else { return true }
        var stack: [Character] = []
    
        for bracket in s {
            switch bracket {
                case "{":
                stack.append("}")
                case "(":
                stack.append(")")
                case "[":
                stack.append("]")
                default:
                    guard bracket == stack.popLast() else { return false }
            }
        }
        return stack.isEmpty
    }
}

File: By-Python/leetcode/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        brackets = []
        
        for item in s:
            if item == "(" or item == "{" or item == "[":
                brackets.append(item)
            else:
                if brackets:
                    if item == ")" and brackets[-1] == "(":
                        brackets.pop()
                    elif item == "}" and brackets[-1] == "{":
                        brackets.pop()
                    elif item == "]" and brackets[-1] == "[":
                        brackets.pop()
                    else:
                        return False
                else:
                    return False
        
        if not brackets:
            return True
        else:
            return False